src/context_dictionary.py

```diff
@@ -21,12 +21,23 @@ def context(document, window_size=10):
         We will assume a default window size of 5 which will be truncated at the
         beginning and end of the sentence.  For example, {'dog': {'cat': 2, 'mouse': 4}}.
     """
-    context_dictionary = defaultdict(Counter)
+    # context_dictionary = defaultdict(Counter)
+    context_dictionary = {}
     for sentence in document:
         for idx, word in enumerate(sentence):
             left_window = sentence[max(0, idx - window_size // 2): idx]
             right_window = sentence[idx + 1: idx + window_size // 2]
-            context_dictionary[word].update(chain(left_window, right_window))
+            word_context = {}
+            for char in chain(left_window, right_window):
```

Comment:
I didn't want to use word again but you make a good point I can just change it to word.